component/src/main/java/org/wso2/extension/siddhi/io/websocket/sink/WebSocketSink.java

```diff
@@ -0,0 +1,201 @@
+                idleTimeout = Integer.parseInt(idleTimeoutString);
+                if (idleTimeout < -1) {
+                    throw new SiddhiAppCreationException("The idle timeout defined in '" + streamDefinition +
+                                                                 "' should be greater than 0.");
+                }
+            } catch (NumberFormatException e) {
+                throw new SiddhiAppCreationException("The idle timeout defined in '" + streamDefinition +
+                                                             "' should be an Integer.");
+            }
+        }
+        connectorListener = new WebSocketClientConnectorListener();
+        try {
+            String scheme = (new URI(url)).getScheme();
+            if (!Objects.equals("ws", scheme) && !Objects.equals("wss", scheme)) {
+                throw new SiddhiAppCreationException("Invalid scheme in " + WebSocketConstants.URL + " = " +
+                                                             url + ". The scheme of the " + WebSocketConstants.URL +
+                                                             " for the websocket server should be either `ws` or "
+                                                             + "`wss`.");
+            }
+        } catch (URISyntaxException e) {
+            throw new SiddhiAppCreationException("There is an syntax error in the '" + url + "' of the websocket "
+                                                         + "server.", e);
+        }
+    }
+
+    @Override
+    public void connect() throws ConnectionUnavailableException {
+        HttpWsConnectorFactoryImpl httpConnectorFactory = new HttpWsConnectorFactoryImpl();
+        WsClientConnectorConfig configuration = new WsClientConnectorConfig(url);
+        if (subProtocol != null) {
+            String[] subProtocol1 = WebSocketUtil.getSubProtocol(subProtocol);
+            configuration.setSubProtocols(subProtocol1);
+        }
+        if (headers != null) {
+            Map<String, String> customHeaders = WebSocketUtil.getHeaders(headers);
+            configuration.addHeaders(customHeaders);
+        }
+        if (idleTimeoutString != null) {
+            configuration.setIdleTimeoutInMillis(idleTimeout);
+        }
+        WebSocketClientConnector clientConnector = httpConnectorFactory.createWsClientConnector(configuration);
+        handshakeFuture = clientConnector.connect(connectorListener);
+    }
+
+    @Override
+    public void publish(Object payload, DynamicOptions dynamicOptions) throws ConnectionUnavailableException {
+        WebSocketSinkHandshakeListener handshakeListener = new WebSocketSinkHandshakeListener
+                (payload, streamDefinition);
+        handshakeFuture.setHandshakeListener(handshakeListener);
+    }
+
+    @Override
+    public void disconnect() {
+    }
+
+    @Override
+    public void destroy() {
+    }
+
+    @Override
+    public Map<String, Object> currentState() {
+        return null;
```

Comment:
Instead lets return `Collections.emptyMap()`

Reply:
fixed in b5c9ede